src/app/Dashboard/LayoutTemplateGroup.tsx

```diff
@@ -0,0 +1,182 @@
+ * Works (as defined below), to deal in both
+ *
+ * (a) the Software, and
+ * (b) any piece of software and/or hardware listed in the lrgrwrks.txt file if
+ * one is included with the Software (each a "Larger Work" to which the Software
+ * is contributed by such licensors),
+ *
+ * without restriction, including without limitation the rights to copy, create
+ * derivative works of, display, perform, and distribute the Software and make,
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+import { dashboardConfigTemplateHistoryClearIntent } from '@app/Shared/Redux/ReduxStore';
+import { ServiceContext } from '@app/Shared/Services/Services';
+import { portalRoot } from '@app/utils/utils';
+import { CatalogTile } from '@patternfly/react-catalog-view-extension';
+import { Button, Dropdown, DropdownItem, Gallery, KebabToggle, Split, SplitItem, Title } from '@patternfly/react-core';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { iconify, LayoutTemplate, LayoutTemplateContext, LayoutTemplateVendor } from './dashboard-utils';
+
+export interface LayoutTemplateGroupProps {
+  title: string;
+  templates: LayoutTemplate[];
+  onTemplateSelect: (templateName: LayoutTemplate) => void;
+  onTemplateDelete: (templateName: string) => void;
+}
+
+export const LayoutTemplateGroup: React.FC<LayoutTemplateGroupProps> = ({
+  onTemplateSelect,
+  onTemplateDelete,
+  ...props
+}) => {
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+  const serviceContext = React.useContext(ServiceContext);
+  const { selectedTemplate, setSelectedTemplate } = React.useContext(LayoutTemplateContext);
+  const scrollRef = React.useRef<HTMLDivElement>(null);
+
+  const handleTemplateSelect = React.useCallback(
+    (template: LayoutTemplate) => {
+      setSelectedTemplate(template);
+      onTemplateSelect(template);
+    },
+    [setSelectedTemplate, onTemplateSelect]
+  );
+
+  const KebabCatalogTileBadge: React.FC<{ template: LayoutTemplate }> = ({ template }) => {
```

Comment:
Extract to top-level const? In current case, a new component type is created anytime this LayoutTemplateGroup builds.